jerry-core/ecma/operations/ecma-array-object.c

```diff
@@ -230,12 +217,13 @@ ecma_op_array_object_set_length (ecma_object_t *object_p, /**< the array object
     current_len_uint32 = ecma_delete_array_properties (object_p, new_len_uint32, old_len_uint32);
   }
 
-  ecma_value_assign_uint32 (&len_prop_value_p->value, current_len_uint32);
+  ext_object_p->u.array.length = current_len_uint32;
```

Comment:
After the conversion to virtual value, it is an uint32. When you get array.length it is converted to ecma_value.